sci-biology/ncbi-blast+/ncbi-blast+-2.10.0.ebuild

```diff
@@ -28,13 +28,14 @@ IUSE="
 	berkdb boost bzip2 cppunit curl expat fltk freetype gif
 	glut gnutls hdf5 icu jpeg lzo mesa mysql muparser opengl pcre png python
 	sablotron sqlite tiff xerces xalan xml xpm xslt X"
-KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+KEYWORDS="~amd64 ~x86"
 #KEYWORDS=""
 
 REQUIRED_USE="${PYTHON_REQUIRED_USE}"
 
 # sys-libs/db should be compiled with USE=cxx
-DEPEND="
+BDEPEND="
+	<sys-devel/gcc-10
```

Comment:
I'm testing this out as we speak and will be pulling it if all goes well, but I'd really recommend you spare some thought whether this could be bumped in the near future so that we have a version working with current GCC

Reply:
Most-likely not. The suite was always developed with mutch more older gcc, I assume still gcc-8. They are slowly implementing a newr build system in `src/build-system/cmake/cmake-configure`. The configure was not rebust enough to automagically enable all tweaks so practically the Mafile.flat approach needs to be used to enable disable a number of useful but optional utiltiies, libs, "plugins".

Subscribe to http://www.ncbi.nlm.nih.gov/mailman/listinfo/cpp , the devs are quick to respond.

Reply:
Fishing out from my old emails, which I copied over to gnetoo bugzilla or github scince in the past, btw:

```
Hi, Martin.

This code needs a variant of GLEW that the 2.0 release dropped:

https://github.com/nigels-com/glew/issues/38

For ncbi-tools++, you might consider simply building --without-gui, but
that's obviously not an option for gbench.

-- Aaron Ucko <ucko at ncbi>, NCBI C++ Toolkit core development group

Martin MOKREJŠ <mmokrejs@xxx> writes:

> Hi Aaron,
>   I hit this issue with ncbi-tools++-12.0.0 but realized same happens with gbench-2.10.7 and 2.11.7. I have glew-2.0.0 installed.
>
>
> 2.11.7
>
> /usr/bin/x86_64-pc-linux-gnu-gcc  -std=gnu11 -fgnu89-inline -c  -Wall -Wno-format-y2k  -pthread -pipe -maes -mpclmul -mpopcnt -mavx -march=native -fPIC  -gdwarf-3     -DNCBI_BYPASS_FL_GL -I/usr/include/freetype2  -I/usr/include/libdrm  -D_DEBUG -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE   -D_MT -D_REENTRANT -D_THREAD_SAFE -I/scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/GCC493-DebugMT64/inc -I/scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/include  /scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/src/gui/opengl/ncbi_12x24.c -o ncbi_12x24.o 
> In file included from /scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/src/gui/opengl/glutbitmap.h:15:0,
>                  from /scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/src/gui/opengl/ncbi_12x24.c:4:
> /scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/include/gui/opengl.h:59:23: error: unknown type name 'GLEWContext'
>  NCBI_GUIOPENGL_EXPORT GLEWContext*  glewGetContext();
>                        ^
> In file included from /scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/src/gui/opengl/glutbitmap.h:15:0,
>                  from /scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/src/gui/opengl/ncbi_10x20.c:4:
> /scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/include/gui/opengl.h:59:23: error: unknown type name 'GLEWContext'
>  NCBI_GUIOPENGL_EXPORT GLEWContext*  glewGetContext();
>                        ^
> make[4]: *** [/scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/GCC493-DebugMT64/build/Makefile.rules:67: ncbi_12x24.o] Error 1
> make[4]: *** Waiting for unfinished jobs....
> make[4]: *** [/scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/GCC493-DebugMT64/build/Makefile.rules:67: ncbi_10x20.o] Error 1
> make[4]: Leaving directory '/scratch/var/tmp/portage/sci-biology/gbench-2.11.7/work/gbench-src-2.11.7/GCC493-DebugMT64/build/gui/opengl'
>
>
>
>
>
>
>
> gbench-2.10.7
>
> /usr/bin/x86_64-pc-linux-gnu-gcc  -std=gnu11 -fgnu89-inline -c  -Wall -Wno-format-y2k  -pthread -pipe -maes -mpclmul -mpopcnt -mavx -march=native -fPIC  -gdwarf-3     -DNCBI_BYPASS_FL_GL -I/usr/include/freetype2  -I/usr/include/libdrm  -D_DEBUG -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE   -D_MT -D_REENTRANT -D_THREAD_SAFE -I/scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/GCC493-DebugMT64/inc -I/scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/include  /scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/src/gui/opengl/ncbi_12x24.c -o ncbi_12x24.o 
> In file included from /scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/src/gui/opengl/glutbitmap.h:15:0,
>                  from /scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/src/gui/opengl/ncbi_12x24.c:4:
> /scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/include/gui/opengl.h:59:23: error: unknown type name 'GLEWContext'
>  NCBI_GUIOPENGL_EXPORT GLEWContext*  glewGetContext();
>                        ^
> make[4]: *** [/scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/GCC493-DebugMT64/build/Makefile.rules:67: ncbi_12x24.o] Error 1
> make[4]: *** Waiting for unfinished jobs....
> In file included from /scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/src/gui/opengl/glutbitmap.h:15:0,
>                  from /scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/src/gui/opengl/ncbi_10x20.c:4:
> /scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/include/gui/opengl.h:59:23: error: unknown type name 'GLEWContext'
>  NCBI_GUIOPENGL_EXPORT GLEWContext*  glewGetContext();
>                        ^
> make[4]: *** [/scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/GCC493-DebugMT64/build/Makefile.rules:67: ncbi_10x20.o] Error 1
> make[4]: Leaving directory '/scratch/var/tmp/portage/sci-biology/gbench-2.10.7/work/gbench-src-2.10.7/GCC493-DebugMT64/build/gui/opengl'
> FAILED: src/gui/opengl/Makefile.gui_opengl.lib
```